### ISSUE TITLE
IP Pools: Project user has no way of knowing which IPs are available

### ISSUE THREAD

**Issue by benjaminleonard** [oxidecomputer/console#2040]
This is marked as advanced, but there's no way of knowing which IPs are available – even if they have system-level permissions.

<img src="https://uploads.linear.app/b7545512-bb0c-4743-adc4-e5797212a353/9bcddf55-815d-487d-bf06-6e2207510873/e472a39b-ebf4-4814-853d-925da545a9a6?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2I3NTQ1NTEyLWJiMGMtNDc0My1hZGM0LWU1Nzk3MjEyYTM1My85YmNkZGY1NS04MTVkLTQ4N2QtYmYwNi02ZTIyMDc1MTA4NzMvZTQ3MmEzOWItZWJmNC00ODE0LTg1M2QtOTI1ZGE1NDVhOWE2IiwiaWF0IjoxNzA5NzU0MzUxLCJleHAiOjMzMjgwMzE0MzUxfQ.Y-YLDvunTxgr-SKayKZo6l6gMOCGBM_dC6xDDe67KQU " alt="image.png" width="505" height="652" />

**Comment by benjaminleonard** [comment]
@FelixMcFelix @david-crespo Any context on the use case for including the IP address field here. Presently the user has no way of knowing which are available. I can make an associated issue on omicron and we can disable IP address in the meantime?

**Comment by david-crespo** [comment]
Yes, I think we should just leave the IP field out of the form for now.

**Comment by FelixMcFelix** [comment]
Agreed, but I'm wondering how far we should go to enable it -- we should probably have an endpoint to show which IP addrs are usable in the default pool at least. Should we extend that to all pools linked to the current silo?

**Comment by david-crespo** [comment]
Yeah, if we have an endpoint like that, I'd say they should be able to fetch that list of available IPs for any pool they have access to. Right now they can fetch the IP pool at [`/v1/ip-pools/{pool}`](https://github.com/oxidecomputer/omicron/blob/94bc6f542efc54a06bf6e17159e3b896a61959e5/nexus/src/external_api/http_entrypoints.rs#L1340-L1346). This endpoint could be something like `/v1/ip-pools/{pool}/available-ips`.

This could get a bit weird because it could be a very long list, right? Like @augustuswm just mentioned something about a very large pool of internal VPN IPs.

**Comment by FelixMcFelix** [comment]
I think we do really need to summarise them into ranges/prefixes, otherwise v6 address ranges will *absolutely* be intractable to handle. 😄

**Comment by charliepark** [comment]
Closing this, as the IP address field is no longer on the form:
<img width="515" height="617" alt="Image" src="https://github.com/user-attachments/assets/2c0271fd-7731-4cf4-bd3b-2f5de51b702e" />
